File: 2090_k_radius_arr_avg/sliding_window.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        int n = nums.size();
        int window_size = 2*k+1;
        vector<int> avg_arr(n, -1);
        if(nums.size() < window_size)
            return avg_arr;
        long long int sum = 0;
        int i = 0;

        for(i = 0; i < 2*k+1; i++)
        {
            sum += nums[i];
        }
        avg_arr[k] = int(sum/window_size);

        for(i = k+1; i < n - k; i++)
        {
            sum = sum - nums[i-k-1];
            sum = sum + nums[i+k];
            avg_arr[i] = int(sum/window_size);
        }

        return avg_arr;

    }
};

File: 2090_k_radius_arr_avg/sliding_window.py
import numpy as np
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        avg_arr = [-1]*n
        window_size = 2*k+1
        if n < window_size:
            return avg_arr
        
        sum1 = 0
        sum1 = sum(nums[0:window_size])

        avg_arr[k] = int(sum1/window_size)
        for i in range(k+1,n-k):
            sum1 = sum1 - nums[i-k-1] + nums[i+k]
            avg_arr[i] = int(sum1/window_size)

        return avg_arr
